wd_comp.c

```diff
@@ -513,8 +502,8 @@ int wd_do_comp_sync2(handle_t h_sess, struct wd_comp_req *req)
 
 			strm_req.src += strm_req.src_len;
 			avail_in -= strm_req.src_len;
-			strm_req.src_len = avail_in;
-		} while (strm_req.src_len > 0);
+			strm_req.src_len = chunk;
+		} while (avail_in > 0);
 
```

Comment:
avail_in meaned single spilt packet avail in size before, but now you change it means stream total avail in.  So there is no need the outer while loop ?